File: fix_router.py
import re

with open('lib/router.dart', 'r') as f:
    content = f.read()

favorites_route = """
    GoRoute(
      parentNavigatorKey: rootNavigatorKey,
      path: '/favorites',
      builder: (_, __) => const FavoritesScreen(),
    ),
"""

if "path: '/favorites'" not in content:
    # insert before '/appearance'
    old = """    GoRoute(
      parentNavigatorKey: rootNavigatorKey,
      path: '/appearance',"""
    content = content.replace(old, favorites_route + old)
    with open('lib/router.dart', 'w') as f:
        f.write(content)
    print("Added /favorites route")
else:
    print("Already has /favorites route")

File: fix_router.dart
import 'dart:io';

void main() {
  final file = File('lib/router.dart');
  var content = file.readAsStringSync();
  
  if (!content.contains('import \'screens/earnings_screen.dart\';')) {
    content = content.replaceFirst('import \'screens/inbox_screen.dart\';', 'import \'screens/inbox_screen.dart\';\nimport \'screens/earnings_screen.dart\';');
  }

  if (!content.contains('GoRoute(path: \'/earnings\'')) {
    final newRoute = '''
      GoRoute(path: '/my-orders', builder: (context, state) => const MyOrdersScreen()),
      GoRoute(path: '/earnings', builder: (context, state) => const EarningsScreen()),
''';
    content = content.replaceFirst('GoRoute(path: \'/my-orders\', builder: (context, state) => const MyOrdersScreen()),', newRoute);
    file.writeAsStringSync(content);
    print('Updated router');
  } else {
    print('Router already updated');
  }
}
